psycopg/psycopg/connection.py

```diff
@@ -79,6 +86,119 @@ class Notify(NamedTuple):
 NoticeHandler: TypeAlias = Callable[[e.Diagnostic], None]
 NotifyHandler: TypeAlias = Callable[[Notify], None]
 
+PipelinePendingResult = Union[
+    None,
+    Tuple[
+        "BaseCursor[Any, Any]",
+        Optional[Tuple[Key, Prepare, bytes]],
+    ],
+]
+
+
+class BasePipeline:
```

Comment:
I am moving the pipeline in a separate module.

Reply:
@dlax I have introduced a `_pipeline` module. However, that way I have "won" the whole code. I am happy if you keep the attribution of most of the pipeline code, so I invite you to --reset-author on that commit and push it to your branch :)

Reply:
Done; thanks.